src/Microsoft.OData.Client/Util.cs

```diff
@@ -597,6 +597,18 @@ internal static Version ToVersion(this ODataProtocolVersion protocolVersion)
             return parameters.ToDictionary(parameter => parameter.Name, parameter => Serializer.GetParameterValue(context, parameter));
         }
 
+        /// <summary>
+        /// Quick check if the <paramref name="input"> is Json.
+        /// Note that this does not check if the input is a well formed Json.
+        /// This can be used to check verify that the string is not xml
+        /// </summary>
+        /// <param name="input">the input string.</param>
+        /// <returns>Return true if the input string can be Json.</returns>
+        public static bool CanBeJson(string input)
+        {
+            return input.Trim().Substring(0, 1).IndexOfAny(new[] { '[', '{' }) == 0;

```

Comment:
What if the JSON starts with whitespace, or contains only a number, string, etc.? i.e. what if it's valid JSON that does not begin with `[` or `{`?

Reply:
Note the `Trim()` function. If it does not begin with `[` or `{` therefore it is not a valid Json object that can be parsed.